libc/stdio_scanf.c

```diff
@@ -0,0 +1,4300 @@
+	free(res3);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff2, "%m[1-9] %m[^1-9] %m[1-9]", &res1, &res2, &res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+	free(res1);
+	free(res2);
+	free(res3);
+}
+
+TEST(stdio_scanf_rest, m_c)
+{
+/* Disabled because of issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/667 */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+
+/* Address sanitizer used on Ubuntu 22.04 fails in such case, because of the following issue:
+ * https://github.com/llvm/llvm-project/issues/61768
+ */
+#ifndef __phoenix__
+	TEST_IGNORE();
+#endif
+
+	const char buff[] = "Lor";
+	char *res1 = NULL, *res2 = NULL, *res3 = NULL;
+
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	TEST_ASSERT_NULL(res1);
+	TEST_ASSERT_NULL(res2);
+	TEST_ASSERT_NULL(res3);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff, "%mc%mc%mc", &res1, &res2, &res3));
+
+	TEST_ASSERT_NOT_NULL(res1);
+	TEST_ASSERT_NOT_NULL(res2);
+	TEST_ASSERT_NOT_NULL(res3);
+
+	TEST_ASSERT_EQUAL_CHAR(buff[0], res1[0]);
+	TEST_ASSERT_EQUAL_CHAR(buff[1], res2[0]);
+	TEST_ASSERT_EQUAL_CHAR(buff[2], res3[0]);
+
+	free(res1);
+	free(res2);
+	free(res3);
+}
+
+
+TEST(stdio_scanf_rest, star)
+{
+	const char *buff = "Lorem Ipsum Dolor 123 SitAmet c 123 0x233";
+	char res1[MAX_TESTSTR_WORDLEN], res;
+	int val;
+
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, fscanf(filep, "%*s %*s %s %d %*s %c %*d %*x", res1, &val, &res));
+	TEST_ASSERT_EQUAL_STRING("Dolor", res1);
+	TEST_ASSERT_EQUAL_INT(123, val);
+	TEST_ASSERT_EQUAL_CHAR('c', res);
+
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vfscanfWrapper(filep, "%*s %*s %s %d %*s %c %*d %*x", res1, &val, &res));
+	TEST_ASSERT_EQUAL_STRING("Dolor", res1);
+	TEST_ASSERT_EQUAL_INT(123, val);
+	TEST_ASSERT_EQUAL_CHAR('c', res);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff, "%*s %*s %s %d %*s %c %*d %*x", res1, &val, &res));
+	TEST_ASSERT_EQUAL_STRING("Dolor", res1);
+	TEST_ASSERT_EQUAL_INT(123, val);
+	TEST_ASSERT_EQUAL_CHAR('c', res);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vsscanfWrapper(buff, "%*s %*s %s %d %*s %c %*d %*x", res1, &val, &res));
+	TEST_ASSERT_EQUAL_STRING("Dolor", res1);
+	TEST_ASSERT_EQUAL_INT(123, val);
+	TEST_ASSERT_EQUAL_CHAR('c', res);
+}
+
+TEST(stdio_scanf_rest, field_width)
+{
+	/* Test ignored because of issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/681 */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+
+	char buff[BUFF_LEN], valStr[BUFF_LEN_STR], str[] = "LoreIpsumDolorSitAmet";
+	int intMax = 2147483647, intMin = -2147483647, valIntMin, valIntMax;
+	float fltMax = 3.40282347e+7F, fltMin = 3.40282347e-4F, valFltMin, valFltMax;
+
+	sprintf(buff, "%s %d %d %f %f", str, intMax, intMin, fltMax, fltMin);
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(5, fscanf(filep, "%4s %*s %5d %*d %5d %*d %5f %*f %f", valStr, &valIntMax, &valIntMin, &valFltMax, &valFltMin));
```

Comment:
"%4s %*s %5d %*d %5d %*d %5f %*f %f" repeats a few times, please use  `const char *format` as in the most of cases above. Look through the rest of code and please replace the similar occurrences.